pkg/test/framework/runtime/api/configurable.go

```diff
@@ -0,0 +1,22 @@
+//  Copyright 2019 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package api
+
+import "istio.io/istio/pkg/test/framework/api/component"
+
+// Configurable interface is implemented by Components that accept Configuration.
+type Configurable interface {
```

Comment:
IIRC the reason for this interface is that we're going to configure components automatically while we're resolving requirements. This interface will also allow the test code to attempt to manually configure a component after it's been created, which will probably not work out so well ... but I guess it's not a huge issue.

Reply:
Yes, that's true, we have the same issue with Close, test code could close the component itself. Let me add a comment to the interface describing that this is for the test framework to call with the config passed with the requirement, and is not meant for tests to call directly.

Reply:
sgtm